examples/step-69/step-69.cc

```diff
@@ -0,0 +1,2926 @@
+
+    BoundaryNormalMap boundary_normal_map;
+
+    SparseMatrix<double>                  lumped_mass_matrix;
+    std::array<SparseMatrix<double>, dim> cij_matrix;
+    std::array<SparseMatrix<double>, dim> nij_matrix;
+    SparseMatrix<double>                  norm_matrix;
+
+  private:
+    const MPI_Comm &mpi_communicator;
+    TimerOutput &   computing_timer;
+
+    SmartPointer<const Discretization<dim>> discretization;
+  };
+
+  // @sect4{The <code>ProblemDescription</code> class}
+  //
+  // The member functions of this class are utility functions specific to
+  // Euler's equations:
+  // - The type alias <code>rank1_type</code> is used for the states
+  //   $\mathbf{U}_i^n$
+  // - The type alias <code>rank2_type</code> is used for the fluxes
+  //   $\mathbb{f}(\mathbf{U}_j^n)$.
+  // - The <code>momentum</code> function extracts $\textbf{m}$
+  //   out of the state vector $[\rho,\textbf{m},E]$ and stores it in a
+  //   <code>Tensor<1, dim></code>.
+  // - The <code>internal_energy</code> function computes $E -
+  //   \frac{|\textbf{m}|^2}{2\rho}$ from a given state vector
+  //   $[\rho,\textbf{m},E]$.
+  //
+  // The purpose of the class members <code>component_names</code>,
+  // <code>pressure</code>, and <code>speed_of_sound</code>, is evident
+  // from their names. We also provide a function
+  // <code>compute_lambda_max</code>, that computes the wave speed estimate
+  // mentioned above, $\lambda_{max}(\mathbf{U},\mathbf{V},\mathbf{n})$,
+  // which is used in the computation of the $d_{ij}$ matrix.
+  //
+  // @note The <code>DEAL_II_ALWAYS_INLINE</code> macro expands to a
+  // (compiler specific) pragma that ensures that the corresponding
+  // function defined in this class is always inlined, i.e., the function
+  // body is put in place for every invocation of the function, and no call
+  // (and code indirection) is generated. This is stronger than the
+  // <code>inline</code> keyword, which is more or less a (mild) suggestion
+  // to the compiler that the programmer things it would be beneficial to
+  // inline the function. <code>DEAL_II_ALWAYS_INLINE</code> should only be
+  // used rarely and with caution in situations such as this one, where we
+  // actually know (due to benchmarking) that inlining the function in
+  // question actually improves performance.
+  //
+  // Finally we note that:
+  //  - This is the only class in this tutorial step that is tied to a
+  //    particular "physics" or "hyperbolic conservation law" (in this
+  //    case Euler's equations). All the other classes are primarily
+  //    "discretization" classes, very much agnostic of the particular physics
+  //    being solved.
+  //  - This is a "pure static" class (the antithesis of a
+  //    "pure virtual" class). It's just a convenient way to wrap-up a
+  //    collection of related methods into a single object. Note that we will
+  //    be able to invoke such methods without without creating an instance of
```

Comment:
```suggestion
  //    be able to invoke such methods without creating an instance of
```

Reply:
done

Reply:
There's no need, by the way, to spam everyone saying "Done" -- just mark the conversation as "resolved" :-)

In fact, github will already do that automatically if you commit my suggestions ;-)

Reply:
Message received, notice taken!